### PR TITLE
HDDS-9816. NPE for access to MXBean during shutdown of OM

### PR DESCRIPTION
**Pull request by sumitagrawl**
## What changes were proposed in this pull request?

while stopping OM, mbean is access to get ratis role, and the omRatisServer is set null after stopping server. As fix, omRatisServer is set at end to avoid if used other place, and added a null check.

This is not have much impact as other failure may happen if ratis server is stopped and accessed. Just avoiding NPE here.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9816

## How was this patch tested?

NA

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 1]
```diff
@@ -3031,7 +3031,7 @@ public String getRatisRoles() {
     List<ServiceInfo> serviceList;
     int port = omNodeDetails.getRatisPort();
     RaftPeer leaderId;
-    if (isRatisEnabled) {
+    if (isRatisEnabled && null != omRatisServer) {
```
Comment (kerneltime) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:3034]: Else is redundant at the end, also change the string to `Ratis disable or server shutting down"?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 2]
```diff
@@ -2260,6 +2259,7 @@ public boolean stop() {
       if (omhaMetrics != null) {
         OMHAMetrics.unRegister();
       }
+      omRatisServer = null;
```
[the remark this answers is not in the document]
Comment (sumitagrawl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:2262]: This has impact for Integration test where we stop and start ratis server, either we need create this ratisServer every startup OR need above code.

image

Please share suggestion if need change for startup logic.

[the remark this answers is not in the document]
Comment (sumitagrawl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:2262]: as discussed, not required

**Comment by sadanand48** [comment]
Thanks @sumitagrawl for the change, @kerneltime for the review.

**Comment by sodonnel** [comment]
This commit has broken compile on master. We either need to revert it or fix it ASAP.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project ozone-manager: Compilation failure
[ERROR] /Users/sodonnell/source/ozone2/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:[3038,16] incompatible types: java.lang.String cannot be converted to java.util.List<java.util.List<java.lang.String>>
```

Probably an overlap with:

> HDDS-9303. Display leader in table and highlight current node in OM web UI (#5311

Has caused the issue as the return type was changed there.

**Comment by adoroszlai** [comment]
Created #5764 to fix the compile error.

**Pull request merged**